test/e2e/query_frontend_test.go

```diff
@@ -596,14 +595,36 @@ func TestRangeQueryShardingWithRandomData(t *testing.T) {
 		{{Name: labels.MetricName, Value: "http_requests_total"}, {Name: "pod", Value: "6"}, {Name: "handler", Value: "/metrics"}},
 	}
 
-	startTime := now.Time().Add(-1 * time.Hour)
-	endTime := now.Time().Add(1 * time.Hour)
-	_, err = e2eutil.CreateBlock(ctx, prom.Dir(), timeSeries, 20, timestamp.FromTime(startTime), timestamp.FromTime(endTime), nil, 0, metadata.NoneFunc)
-	testutil.Ok(t, err)
-	testutil.Ok(t, e2e.StartAndWaitReady(prom, sidecar))
+	// Ensure labels are ordered.
+	for _, ts := range timeSeries {
+		sort.Slice(ts, func(i, j int) bool {
+			return ts[i].Name < ts[j].Name
+		})
+	}
+
+	samplespb := make([]prompb.TimeSeries, 0, len(timeSeries))
+	for _, labels := range timeSeries {
+		labelspb := make([]prompb.Label, 0, len(labels))
+		for _, label := range labels {
+			labelspb = append(labelspb, prompb.Label{
+				Name:  string(label.Name),
+				Value: string(label.Value),
+			})
+		}
+		samplespb = append(samplespb, prompb.TimeSeries{
+			Labels: labelspb,
+			Samples: []prompb.Sample{
+				{
+					Value:     float64(0),
+					Timestamp: timestamp.FromTime(predefTimestamp.Time()),
```

Comment:
For the Timestamp of samples, should I have different timestamps from start to end, increment with step size (its implemented like that in CreateBlock function), it worked for the range query but not for instant query in the other test func, so should I add that?

Reply:
I think it doesnt matter for instant query, i might be wrong though!